### PR TITLE
backport run_exports

### PR DESCRIPTION
**Pull request by billysuh7**
Checklist
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

Fixes https://github.com/conda-forge/cuda-cupti-feedstock/issues/14

### CODE REVIEW

**Comment by billysuh7** [comment]
@conda-forge-admin, please rerender

**Comment by billysuh7** [comment]
@conda-forge-admin, please rerender

**Comment by jakirkham** [comment]
Thanks Billy! 🙏

The re-rendering issues are the same ones we discussed recently ( https://github.com/conda-forge/conda-smithy/issues/2011 ). Michael started an upstream fix PR ( https://github.com/conda/conda-build/pull/5458 )

For now, would just do this locally using `conda-build` version `24.5` (any patch release should be fine)

Also in terms of CUDA 12.0, it looks like we updated to CUDA 12.1 in PR ( https://github.com/conda-forge/cuda-cupti-feedstock/pull/4 ). The commit right before that appears to be commit ( https://github.com/conda-forge/cuda-cupti-feedstock/commit/d1f990e47483b84e558efabbbdb02fc3f81b89b3 ). Would it be alright if we use that as the starting point for the `v12.0` branch?

**Comment by jakirkham** [comment]
> Also in terms of CUDA 12.0, it looks like we updated to CUDA 12.1 in PR ( https://github.com/conda-forge/cuda-cupti-feedstock/pull/4 ). The commit right before that appears to be commit ( https://github.com/conda-forge/cuda-cupti-feedstock/commit/d1f990e47483b84e558efabbbdb02fc3f81b89b3 ). Would it be alright if we use that as the starting point for the `v12.0` branch?

Went ahead and made this change. Took the following steps:

1. Renamed the `conda-forge` branch `v12.0` to `v12.0-previous`
2. GitHub automatically reflected that in this PR
3. Created a new `v12.0` branch at commit: https://github.com/conda-forge/cuda-cupti-feedstock/commit/d1f990e47483b84e558efabbbdb02fc3f81b89b3
4. Retargeted this PR to the new `v12.0` branch
5. Deleted the `v12.0-previous` branch

As the new `v12.0` branch is merely a few commits ahead of the old one, it should be possible to fast-forward merge local `v12.0` branches

This PR had a few commits that are now considered included in `v12.0` ( namely those from the long since merged PR: https://github.com/conda-forge/cuda-cupti-feedstock/pull/4 ). After the change above this PR only shows the new changes that are not in `v12.0`. So after these changes this PR needs no updates

So this comment is merely informative. No action needed

**Comment by jakirkham** [comment]
It looks like Travis CI is failing on `linux_ppc64le`. Unfortunately this is a known issue that extends beyond this feedstock or conda-forge. For more context, please see issue: https://github.com/conda-forge/status/issues/185

As to resolving this, we had done some work in PR ( https://github.com/conda-forge/cuda-cupti-feedstock/pull/12 ) to switch to cross-compilation-based builds. Think we should backport that as well to fix CI

The `{{ stdlib("c") }}` issue mentioned by the linter above ( https://github.com/conda-forge/cuda-cupti-feedstock/pull/20#issuecomment-2289831494 ) was fixed in PR: https://github.com/conda-forge/cuda-cupti-feedstock/pull/13

So in short, think we should be backporting these 3 PRs here:

* https://github.com/conda-forge/cuda-cupti-feedstock/pull/10
* https://github.com/conda-forge/cuda-cupti-feedstock/pull/12
* https://github.com/conda-forge/cuda-cupti-feedstock/pull/13

This should fix the CI issues and apply `run_exports`

**Comment by billysuh7** [comment]
@jakirkham thanks a million for all your help!

**Comment by jakirkham** [comment]
Happy to help Billy! Also thank you for working on this 🙏

Let's see what Alex thinks 🙂

**Review by adibbley — APPROVED** [review]

**Comment by jakirkham** [comment]
Thanks Billy and Alex! 🙏
